R/dvhydrograph-data.R

```diff
@@ -1,251 +1,138 @@
-    type <- stat2Timeseries[['type']]
-    logAxis <- isLogged(stat2Timeseries[['points']], stat2Timeseries[['isVolumetricFlow']], excludeZeroNegativeFlag)
-    approvalSeries <- stat2Timeseries
-  } else if(!isEmptyOrBlank(stat3Timeseries)){
-    type <- stat3Timeseries[['type']]
-    logAxis <- isLogged(stat3Timeseries[['points']], stat3Timeseries[['isVolumetricFlow']], excludeZeroNegativeFlag)
-    approvalSeries <- stat3Timeseries
-  } else {
-    type <- ""
-    logAxis <- FALSE
-    approvalSeries <- list()
+  if(isEmptyOrBlank(timeSeries) || !anyDataExist(timeSeries[['points']])){
+    return(NULL)
   }
 
-  #Get max and min IV points
-  max_iv <- getMinMaxIV(reportObject, "MAX", timezone, type, invertedFlag)
-  min_iv <- getMinMaxIV(reportObject, "MIN", timezone, type, invertedFlag)
-
-  #If we are excluding min/max points or if we are excluding zero / negative
-  #points and the max/min vlaues are zero / negative, then replace them
-  #with labels that go on the top of the chart.
-
-  #Max Checking
-  if( (!isEmptyOrBlank(excludeMinMaxFlag) && excludeMinMaxFlag) || 
-      (!isEmptyOrBlank(excludeZeroNegativeFlag) && excludeZeroNegativeFlag && !isEmptyOrBlank(max_iv$value) && max_iv$value <= 0)){
-    max_iv_label <- max_iv
-    not_include <- c(not_include, 'max_iv')
-  } 
-
-  #Min Checking
-  if( (!isEmptyOrBlank(excludeMinMaxFlag) && excludeMinMaxFlag) 
-     || (!isEmptyOrBlank(excludeZeroNegativeFlag) && excludeZeroNegativeFlag && !isEmptyOrBlank(min_iv$value) && min_iv$value <= 0) ){
-    min_iv_label <- min_iv
-    not_include <- c(not_include, 'min_iv')
-  }
+  return(timeSeries)
+}
 
-  #Approvals
-  approvals <- readApprovalBar(approvalSeries, timezone, legend_nm=approvalSeries[['legend.name']], snapToDayBoundaries=TRUE)
+#' Parse DV Approvals
+#'
+#' @description Given a time series, read and format the approvals
+#' data from it into apprpoval bars to put on the plot.
+#' @param timeSeries the time series to get approvals for
+#' @param timezone the timezone to parse the approval times into
+parseDVApprovals <- function(timeSeries, timezone){
+  return(readApprovalBar(timeSeries, timezone, legend_nm=timeSeries[['legend.name']], snapToDayBoundaries=TRUE))
+}
 
-  #Field Visit Measurements
+#' Parse DV Field Visit Measurements
+#'
+#' @description Given the full report JSON object, reads the field
+#' visit measurements and handles read errors.
+#' @param reportObject the full report JSON object
+parseDVFieldVisitMeasurements <- function(reportObject){
   meas_Q <- tryCatch({
     readFieldVisitMeasurementsQPoints(reportObject)
   }, error = function(e) {
     warning(paste("Returning empty data frame as DV Hydro field visit measurements. Error:", e))
-    na.omit(data.frame(time=as.POSIXct(NA), value=as.numeric(NA), minQ=as.numeric(NA), maxQ=as.numeric(NA), n=as.numeric(NA), month=as.character(NA), stringsAsFactors=FALSE))
+    return(NULL)
   })
-  
-  #Ground Water Levels
+
+  if(!anyDataExist(meas_Q)){
```

Comment:
I don't think it's as applicable here because I believe the field visit measurements and ground water level arrays will never be filled if there are no points, while a time series field could be. Probably better to be safe and include it though, so I will. 👍